js/components/contextMenu.js

```diff
@@ -252,12 +237,165 @@ class ContextMenuSingle extends ImmutableComponent {
  * Represents a context menu including all submenus
  */
 class ContextMenu extends ImmutableComponent {
+  constructor () {
+    super()
+    this.onKeyDown = this.onKeyDown.bind(this)
+  }
+
+  componentDidMount () {
+    window.addEventListener('keydown', this.onKeyDown)
+    window.addEventListener('keyup', this.onKeyUp)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('keydown', this.onKeyDown)
+    window.removeEventListener('keyup', this.onKeyUp)
+  }
+
+  onKeyDown (e) {
+    const selectedIndex = this.props.selectedIndex
+    const selectedTemplate = this.getMenuByIndex(selectedIndex, this.props.contextMenuDetail.get('template'))
+    const selectedMenuItem = selectedTemplate.get(this.currentIndex)
+
+    switch (e.keyCode) {
+      case keyCodes.ENTER:
+        e.preventDefault()
+        e.stopPropagation()
+        const action = selectedTemplate.getIn([this.currentIndex, 'click'])
+        if (action) {
+          action(e)
+        }
+        windowActions.resetMenuState()
+        break
+
+      case KeyCodes.ESC:
+      case KeyCodes.TAB:
+        windowActions.resetMenuState()
+        break
+
+      case keyCodes.LEFT:
+        if (e.which === keyCodes.LEFT && this.hasSubmenuSelection) {
```

Comment:
because the code is complex, if you could keep the original comments in there, it would be much appreciated :smile: For example, this one had:

```
// Left arrow inside a submenu
// <= go back one level
```